ui/src/pages/_app.tsx

```diff
@@ -15,13 +15,24 @@ import '../styles/globals.css';
 
 type CustomProps = {
     protectedPage?: boolean;
+    metadata?: {
+        title?: string;
+        description?: string;
+    };
 };
 
 const App = ({ Component, pageProps }: Types.AppProps<CustomProps>) => {
     const [mounted, setMounted] = useState(false);
     const { user } = Stores.useAuthStore();
     const { darkMode } = Stores.usePreferencesStore();
 
+    const metadata = {
```

Comment:
The code inside _app.tsx will execute on every route/subroute change, even when adding query params into the URL. I don't recommend adding code into this file unless it's absolutely necessary. I recommend leaving this responsibility (of rendering proper meta tags) at the page level as it was. Since you already added that "metadata" prop which is computed server side, the tags can be rendered with the info coming via props on each page.